site/app/controllers/student/TeamController.php

```diff
@@ -417,9 +417,35 @@ public function showPage($gradeable_id) {
         }
 
         $teams = $this->core->getQueries()->getTeamsByGradeableId($gradeable_id);
+
+        $members = [];
+        $seekers = [];
+        $invites_received = [];
+        $users_seeking_team = $this->core->getQueries()->getUsersSeekingTeamByGradeableId($gradeable_id);
+        $seeking_partner = false;
+        if ($team !== null) {
+            //List team members
+            foreach ($team->getMembers() as $teammate) {
+                $members[] = $this->core->getQueries()->getUserById($teammate);
+            }
+        } else {
+            //Invites
+            foreach($teams as $t) {
+                if ($t->sentInvite($user_id)) {
+                    $invites_received[] = $t;
+                }
+            }
+
+            //Are you seeking a team?
+            $seeking_partner = in_array($user_id, $users_seeking_team);
+        }
+
+        foreach ($users_seeking_team as $user_seeking_team) {
+            $seekers[] = $this->core->getQueries()->getUserById($user_seeking_team);
+        }
+
```

Comment:
Is the reason you have so much code moved from the view to the controller because it's all logic that is imperative to the query that 'fills' the $seekers[] array?

Reply:
It looks like there are multiple queries contained within this code block, thus necessitating its migration to the controller.

Reply:
It just seemed cleaner and easier to read leaving all the logic together. Like I could put the line for $seeking_partner  in the view but then you have to pass $users_seeking_team to the view and that didn't seem to improve anything.